File: new-resource-detector/src/main/java/nl/haploid/octowight/detector/ResourceDetector.java
package nl.haploid.octowight.detector;

import nl.haploid.octowight.AtomChangeEvent;
import nl.haploid.octowight.registry.data.ResourceRoot;

import java.util.Collection;
import java.util.List;

public interface ResourceDetector { // TODO: rename?

	Collection<String> getAtomTypes();

	List<ResourceRoot> detect(final List<AtomChangeEvent> events);
}

File: new-resource-detector/src/main/scala/nl/haploid/octowight/detector/ResourceDetector.scala
package nl.haploid.octowight.detector

import nl.haploid.octowight.AtomChangeEvent
import nl.haploid.octowight.registry.data.ResourceRoot

trait ResourceDetector {

  def getAtomCategories: List[String]

  def detect(events: Traversable[AtomChangeEvent]): Traversable[ResourceRoot]
}
